src/Components/Endpoints/src/DependencyInjection/PersistedStateSerializationMode.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.Components;
+
+/// <summary>
+/// Specifies the mode to use when serializing component persistent state.
+/// </summary>
+public enum PersistedStateSerializationMode
+{
+    /// <summary>
+    /// Indicates that the serialization mode should be inferred from the current request context.
+    /// </summary>
+    Infer = 1,
+
+    /// <summary>
+    /// Indicates that the state should be persisted so that execution may resume on Server.
+    /// </summary>
+    Server = 2,
+
+    /// <summary>
+    /// Indicates that the state should be persisted so that execution may resume on WebAssembly.
+    /// </summary>
+    WebAssembly = 3,
+}
```

Comment:
Are we not going to require serializing state both for server and client in a single request?

Reply:
Is that what infer means?

Reply:
Infer is the same as what we had before when you don't specify any render mode, and it gets inferred from the render modes used within the request.

> Are we not going to require serializing state both for server and client in a single request?

Yes, this will definitely need to change later once we support emitting both server and wasm interactive components. For now I'm retaining the existing functionality through this refactoring rather than implementing substantial new functionality.